src/test/scala/com/lightbend/lagom/raffle/model/Raffle.scala

```diff
@@ -0,0 +1,144 @@
+package com.lightbend.lagom.raffle.model
+
+import java.time.OffsetDateTime
+
+import akka.Done
+import com.lightbend.lagom.core.es.PersistentEntity
+import com.lightbend.lagom.core.es.PersistentEntity.ReplyType
+
+import scala.util.Random
+
+object Raffle extends PersistentEntity {
+
+  type State = Raffle
+  type Command = RaffleCommand
+  type Event = RaffleEvent
+
+  override def behavior =
+    Behavior
+      .initialState(EmptyRaffle)
+      .update {
+
+        case EmptyRaffle =>
+          EmptyRaffle.acceptParticipants and
+            EmptyRaffle.canNotRunWithoutParticipants and
+            EmptyRaffle.ignoreRemovals
+
+        case raffle: NonEmptyRaffle =>
+          raffle.acceptParticipants and
+            raffle.removeParticipants and
+            raffle.run
+
+        case FinishedRaffle =>
+          FinishedRaffle.rejectAllCommands
+      }
+}
+
+import com.lightbend.lagom.raffle.model.Raffle._
+
+sealed trait Raffle
+
+case object EmptyRaffle extends Raffle {
+
+
+  val acceptParticipants =
+    actions
```

Comment:
I think that having to `import com.lightbend.lagom.raffle.model.Raffle` to use `actions` here is not very intuitive. What do you think about making the whole ADT (`Raffle`, `EmtpyRaffle`, ...) become inner classes of the `object Raffle extends PersistentEntity`? Something like:

```scala
object Raffle extends PersistentEntity {

  type State = Raffle
  type Command = RaffleCommand
  type Event = RaffleEvent

  override def behavior = ...

  sealed trait Raffle
  case object EmptyRaffle extends Raffle { ... }
  case class NonEmptyRaffle(participants: List[String]) extends Raffle { ... }
  case object FinishedRaffle extends Raffle { ... }
}
```

Then I think one of the `Raffle` _thingies_ (the persistent entity or the trait) could be renamed. Does that make sense? As in, one is the `RaffleEntity` and the other is `RaffleState`, then we'd have:


```scala
object RaffleEntity extends PersistentEntity { // don't really like the name 'RaffleEntity' :-(

  type State = RaffleState
  type Command = RaffleCommand
  type Event = RaffleEvent

  override def behavior = ...

  // this can lead to confusion because it's a super trait for all possible states in the FSM and also
  // the state as we know it in a CRUD application. For example:
  //     NonEmptyRaffle(Nil) and NonEmptyRaffle(List("Ignasi") 
  // are two different states in the same state of the FSM.  Effect.persist( new MindExploded( mine ) )
  sealed trait RaffleState   
  case object EmptyRaffle extends RaffleState { ... }
  case class NonEmptyRaffle(participants: List[String]) extends RaffleState { ... }
  case object FinishedRaffle extends RaffleState { ... }
}
```

Reply:
I will try it out. 

In any case, that's not something that we can (or should) enforce. Users will be free to organise their code the way they prefer. 

But if it increases the readability of our samples, yes, I'm totally for it.